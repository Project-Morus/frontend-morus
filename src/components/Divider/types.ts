export interface IDividerProps {
  maxWidth?: number;
  marginTop?: number;
  marginBottom?: number;
  variant: VariantsTypes;
}

export type VariantsTypes = "primary" | "secondary";
