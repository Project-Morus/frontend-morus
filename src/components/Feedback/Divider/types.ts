import { IBaseProps } from "../../../types";

export interface IDividerProps extends IBaseProps {
  maxWidth?: number;
  marginTop?: string;
  marginBottom?: string;
  variant?: VariantsTypes;
}

export type VariantsTypes = "primary" | "secondary";
