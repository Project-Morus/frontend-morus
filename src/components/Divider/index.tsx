import { ReactNode } from "react";
import { IDividerProps, VariantsTypes } from "./types";
import DividerPrimary from "./variants/Primary";
import DividerSecondary from "./variants/Secondary";

const Divider = ({ variant = "primary", ...props }: IDividerProps) => {
  const variantType: Record<VariantsTypes, ReactNode> = {
    primary: <DividerPrimary {...props} />,
    secondary: <DividerSecondary {...props} />,
  };

  return variantType[variant];
};

export default Divider;
