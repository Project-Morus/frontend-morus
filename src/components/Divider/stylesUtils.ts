import styled from "styled-components";

export const BASE_STYLE = styled.div<{
  $maxWidth?: number;
  $marginTop?: string;
  $marginBottom?: string;
}>`
  width: 100%;

  max-width: ${({ $maxWidth }) => $maxWidth && $maxWidth}px;

  margin-bottom: ${({ $marginBottom }) => $marginBottom && $marginBottom};
  margin-top: ${({ $marginTop }) => $marginTop && $marginTop};

  border-width: ${({ theme }) => theme.border.width.regular};
  border-style: solid;
`;
