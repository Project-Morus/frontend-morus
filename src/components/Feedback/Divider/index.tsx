import { ReactNode } from "react";
import { IDividerProps, VariantsTypes } from "./types";
import { DividerPrimary, DividerSecondary } from "./variants";

const Divider = ({ variant = "primary", ...props }: IDividerProps) => {
  const variantType: Record<VariantsTypes, ReactNode> = {
    primary: <DividerPrimary {...props} role='divider' />,
    secondary: <DividerSecondary {...props} role='divider' />,
  };

  return variantType[variant];
};

export default Divider;
