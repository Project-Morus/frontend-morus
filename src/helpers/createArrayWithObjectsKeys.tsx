export const createArrayWithObjectsKeys = (quantity: number) => {
  const dotsKeys = []

  for (let i = 1; i <= quantity; i++) {
    [...dotsKeys, dotsKeys.push({ key: i })]
  }

  return dotsKeys
}