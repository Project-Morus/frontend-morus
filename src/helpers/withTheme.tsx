import { ComponentType, JSX } from 'react';
import { ThemeProvider } from "styled-components";
import { defaultTheme } from "../styles/themes/default";

type ComponentPropsType<T> = T & JSX.IntrinsicAttributes

export function withTheme<T>(WrapperComponent: ComponentType<ComponentPropsType<T>>) {
  return (props: ComponentPropsType<T>) => {
    return (
      <ThemeProvider theme={defaultTheme}>
        <WrapperComponent {...props} />
      </ThemeProvider>
    );
  };
}

