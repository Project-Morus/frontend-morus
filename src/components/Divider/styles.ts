import { styled } from "styled-components";
import { BASE_STYLE } from "./stylesUtils";

export const DividerPrimarySC = styled(BASE_STYLE)`
  border: ${({ theme }) => theme.border.width.bold} solid ${({ theme }) => theme.colors.primary[50]};
`;

export const DividerSecondarySC = styled(BASE_STYLE)`
  border: ${({ theme }) => theme.border.width.bold} solid ${({ theme }) => theme.colors.grey[75]};
`;
