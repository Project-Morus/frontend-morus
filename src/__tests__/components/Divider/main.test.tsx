import { render, screen } from '@testing-library/react'
import { withTheme } from '../../../helpers/withTheme'
import Divider from '../../../components/Divider'

const DividerWithTheme = withTheme(Divider)
describe('Divider Component', () => {
  test('Verify if component is rendering', () => {
    render(<DividerWithTheme />)
    expect(screen.getByRole('divider')).toBeVisible()
  })

  test('Snapshot Divider', () => {
    const { asFragment } = render(<DividerWithTheme />)

    expect(asFragment()).toMatchSnapshot()
  })
  test('Snapshot Divider with variant secondary', () => {
    const { asFragment } = render(<DividerWithTheme variant='secondary' />)

    expect(asFragment()).toMatchSnapshot()
  })
})