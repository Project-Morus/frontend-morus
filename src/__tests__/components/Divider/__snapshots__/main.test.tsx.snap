// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Divider Component Snapshot Divider 1`] = `
<DocumentFragment>
  .c0 {
  width: 100%;
  max-width: px;
  border-width: 1px;
  border-style: solid;
}

.c1 {
  border-color: #E6EEF6;
}

<div
    class="c0 c1"
    role="divider"
  />
</DocumentFragment>
`;

exports[`Divider Component Snapshot Divider with variant secondary 1`] = `
<DocumentFragment>
  .c0 {
  width: 100%;
  max-width: px;
  border-width: 1px;
  border-style: solid;
}

.c1 {
  border-color: #D9D9D9;
}

<div
    class="c0 c1"
    role="divider"
  />
</DocumentFragment>
`;
